lambdas/sharedattributes/src/test/java/uk/gov/di/ipv/core/sharedattributes/SharedAttributesHandlerTest.java

```
@@ -123,36 +157,30 @@ void shouldExtractSessionIdFromHeaderAndReturnSharedAttributesAndStatusOK()
+        assertEquals(3, claims.get(VC_HTTP_API_CLAIM).size());
         JsonNode vcAttributes = claims.get(VC_HTTP_API_CLAIM);
 
-        assertEquals(2, (vcAttributes.get("names")).size());
-
-        vcAttributes
-                .get("names")
-                .forEach(
-                        (name) -> {
-                            if (name.get("familyName").asText().equals("Holmes")) {
-                                assertEquals(
-                                        "[ \"Sherlock\" ]",
-                                        name.get("givenNames").toPrettyString());
-                            } else if (name.get("familyName").asText().equals("Watson")) {
-                                assertEquals(
-                                        "[ \"John\", \"H\" ]",
-                                        name.get("givenNames").toPrettyString());
-                            } else {
-                                fail("Unexpected familyName");
-                            }
-                        });
-
-        assertEquals("2021-03-01", vcAttributes.get("dateOfBirths").get(0).asText());
-        assertTrue(vcAttributes.get("addresses").toPrettyString().contains("123 Street"));
-        assertEquals(2, vcAttributes.get("addresses").size());
-        assertEquals(2, vcAttributes.get("addressHistory").size());
+        JsonNode address = vcAttributes.get("address");
+        if (address != null) {
+            for (JsonNode jo : address) {
+                if (jo.get("streetAddress") != null) {
+                    if (jo.get("streetAddress").asText().equals("758 Huel Neck")) {
+                        assertEquals("758 Huel Neck", jo.get("streetAddress").asText());
+                    }
+                    if (jo.get("postalCode").asText().equals("S5 6UN")) {
+                        assertEquals("S5 6UN", jo.get("postalCode").asText());
+                    }
+                }
+            }
+        }
```

I think this can be simplified. You can probably drop the null checks - this is just a test so we should be confident in the data we're working with. 
Also, you've got if statements doing a pre-check on the assertions. So you're tests will silently fail if the assertion is wrong.

---

There are 3 addresses blocks , one of them does not have streetaddress (ie set to null & is not shown) . This is why I need to check if the address is not null . Also the address blocks are not always displayed in the same order . I cannot say for ex , assert that block address[1].streetname equala "758.." . To get round , I check if the  streetaddress.exists & therefore assert ..